driver-dse/src/main/java/com/datastax/driver/graph/BoundGraphStatement.java

```diff
@@ -0,0 +1,116 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.graph;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.exceptions.DriverException;
+
+/**
+ * Bound graph statement produced from a prepared statement.
+ */
+public class BoundGraphStatement extends AbstractGraphStatement<BoundStatement> {
+
+    //  "name", "value"
+    Map<String, Object> valuesMap;
+
+    BoundGraphStatement(BoundStatement bs, GraphStatement gst) {
+        super(gst.getSession());
+        this.wrappedStatement = bs;
+        this.valuesMap = new HashMap<String, Object>();
+
+        /*
+        Need to keep the configuration from the Prepared statement to
+        the created BoundStatement, so users don't have to re configure the payload and such.
+         */
+        copyConfigFromStatement(gst);
+    }
+
+    void copyConfigFromStatement(GraphStatement gst) {
+        this.wrappedStatement.setOutgoingPayload(gst.getGraphOptions() == null ? this.session.getDefaultGraphOptions() : gst.getGraphOptions());
+    }
+
+    // Bind variables in the PreparedStatement
+    @Override
+    BoundStatement configureAndGetWrappedStatement() {
+        JsonNodeFactory factory = new JsonNodeFactory(false);
+        JsonFactory jsonFactory = new JsonFactory();
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
ObjectMapper instances are thread-safe and should be shared; they can even be declared as singletons.